Markdown/Markdown.sublime-syntax

```diff
@@ -180,7 +180,21 @@ variables:
     #     \s*$      # any amount of whitespace until EOL
     #   )
 contexts:
+  file-start:
+    - match: ^---$\n
```

Comment:
Using `^` here might cause some issues if Markdown is embedded in another syntax for some reason. Using it here would also mean to use it in the push statement for `default`.

If `\n` is matched without the need to detect the end-of-file, the `$` before can be omitted.

Result: 
```
- match: ---\n
```